### DISCUSSION TITLE
MySQL TINYINT - not able to set both unsigned and length

### DISCUSSION THREAD

**Discussion by brujah** [sqlalchemy/sqlalchemy#9964] [Usage Questions]
```my_db_column: Mapped[bool] = mapped_column(TINYINT(unsigned=True), nullable=False, default=0```
This one makes it unsigned but with length of 0

```my_db_column: Mapped[bool] = mapped_column(TINYINT(1), nullable=False, default=0```
This one does not make it unsigned but with a length of 1

```my_db_column: Mapped[bool] = mapped_column(TINYINT(1, unsigned=True), nullable=False, default=0```
This one does makes it unsigned but ignores the length and sets it to 0

**Comment by zzzeek** [comment]
hi - 

can't reproduce.  try this out

```py
>>> from sqlalchemy.dialects.mysql import TINYINT
>>> from sqlalchemy.dialects import mysql

>>> print(TINYINT(unsigned=True).compile(dialect=mysql.dialect()))
TINYINT UNSIGNED

>>> print(TINYINT(1).compile(dialect=mysql.dialect()))
TINYINT(1)

>>> print(TINYINT(1, unsigned=True).compile(dialect=mysql.dialect()))
TINYINT(1) UNSIGNED

```

**Reply by zzzeek** [reply]
test case

```py
from __future__ import annotations

from sqlalchemy import create_engine
from sqlalchemy.dialects.mysql import TINYINT
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column


class Base(DeclarativeBase):
    pass


class A(Base):
    __tablename__ = "a"

    id: Mapped[int] = mapped_column(primary_key=True)
    my_db_column: Mapped[bool] = mapped_column(
        TINYINT(1, unsigned=True), nullable=False, default=0
    )


e = create_engine("mysql://scott:tiger@mysql80/test", echo=True)
Base.metadata.create_all(e)

```

output

```sql
CREATE TABLE a (
	id INTEGER NOT NULL AUTO_INCREMENT, 
	my_db_column TINYINT(1) UNSIGNED NOT NULL, 
	PRIMARY KEY (id)
)

```

LGTM

**Reply by brujah** [reply]
Hi!

Yes all of your code looks the same when I run it on my machine but when I open up the column in design mode (using Navicat) the length of that column is still without value

**Reply by brujah** [reply]
Seems more like a MySQL thing as if i run the query directly in Navicat the column does not get a length even though it is specific in the query with ```TINYINT(2)```

**Reply by zzzeek** [reply]
yes, MySQL likes to make choices of its own without telling you.    the syntax seems supported https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html so i dont have any further info on what it's deciding.
